File: apps/api/app/schemas/planning.py
from datetime import datetime

from pydantic import BaseModel, Field

from app.models.enums import WorkStatus


class IdeaCreate(BaseModel):
    title: str = Field(min_length=1, max_length=180)
    body: str | None = Field(default=None, max_length=10_000)
    score: int = Field(default=0, ge=0, le=100)
    source: str | None = Field(default=None, max_length=80)


class IdeaRead(IdeaCreate):
    id: str
    workspace_id: str
    project_id: str
    created_at: datetime


class JournalCreate(BaseModel):
    title: str = Field(min_length=1, max_length=180)
    body: str = Field(min_length=1, max_length=50_000)
    mood: str | None = Field(default=None, max_length=40)


class JournalRead(JournalCreate):
    id: str
    workspace_id: str
    project_id: str
    summary: str | None
    created_at: datetime


class MilestoneCreate(BaseModel):
    title: str = Field(min_length=1, max_length=180)
    description: str | None = Field(default=None, max_length=10_000)
    due_date: datetime | None = None


class MilestoneUpdate(BaseModel):
    title: str | None = Field(default=None, min_length=1, max_length=180)
    description: str | None = Field(default=None, max_length=10_000)
    status: WorkStatus | None = None
    due_date: datetime | None = None


class MilestoneRead(BaseModel):
    id: str
    workspace_id: str
    project_id: str
    title: str
    description: str | None
    status: WorkStatus
    due_date: datetime | None
    completed_at: datetime | None
    created_at: datetime

File: apps/web/src/types/planning.ts
import type { WorkStatus } from "@/types/domain";

export interface PlanningProject {
  id: string;
  name: string;
  codename: string;
}

export interface ProjectIdea {
  id: string;
  projectId: string;
  title: string;
  body?: string;
  score: number;
  source?: string;
  createdAt: string;
}

export interface JournalRecord {
  id: string;
  projectId: string;
  title: string;
  body: string;
  mood?: string;
  summary?: string;
  createdAt: string;
}

export interface ProjectMilestone {
  id: string;
  projectId: string;
  title: string;
  description?: string;
  status: WorkStatus;
  dueDate?: string;
  completedAt?: string;
  createdAt: string;
}
